Core AMQP cleanup for v2 (#12210)
Azure/azure-sdk-for-js: sdk/core/core-amqp/CHANGELOG.md

- `AmqpAnnotatedMessage` interface that closely represents the AMQP annotated message from the [AMQP spec](https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#section-message-format) has been added. New `AmqpMessageHeaders` and `AmqpMessageProperties` interfaces(properties with camelCasing) have been added in the place of re-exports from "rhea" library(properties with snake_casing).
  [PR 12091](https://github.com/Azure/azure-sdk-for-js/pull/12091)

### Breaking changes

We are cleaning the public API surface by

> - removing exports that are either not used by either `@azure/event-hubs` and `@azure/service-bus` packages (which are the two main consumers of this package)

Reviewer: NIT
```suggestion
- removing exports that are neither used by `@azure/event-hubs` nor `@azure/service-bus` package (which are the two main consumers of this package)
```
